packages/router/router.ts

```diff
@@ -3361,34 +3370,38 @@ function getMatchesToLoad(
       return;
     }
 
+    // Revalidating fetchers are decoupled from the route matches since they
+    // load from a static href.  They only set `defaultShouldRevalidate` on
+    // explicit revalidation due to submission, useRevalidator, or X-Remix-Revalidate
+    //
+    // They automatically revalidate without even calling shouldRevalidate if:
+    // - They were cancelled
+    // - They're in the middle of their first load and therefore this is still
+    //   an initial load and not a revalidation
+    //
+    // If neither of those is true, then they _always_ check shouldRevalidate
+    let fetcher = state.fetchers.get(key);
+    let isPerformingInitialLoad =
+      fetcher &&
+      fetcher.state !== "idle" &&
+      fetcher.data === undefined &&
+      // If a fetcher.load redirected then it'll be "loading" without any data
+      // so ensure we're not processing the redirect from this fetcher
+      !fetchRedirectIds.has(key);
```

Comment:
This is the net new condition, otherwise this code just combines the if blocks for `cancelledFetcherLoads`/`isPerformingInitialLoad`/`shouldRevalidate` since they all populate the same object into `revalidatingFetchers`.